src/on_chain/on_chain_daemon.py

```diff
@@ -0,0 +1,56 @@
+"""
+At runtime, "main" function records the most recent block and initializes it as the start block.
+After the daemon is asleep for 30 mins, it gets the newest block as the end block.
+Since the competition endpoint has a lag of 30 mins (worst case), we wait 30 mins before we
+fetch comp. data and start checking for potential surplus.
+If the data was not retrievable for any reason, it adds it to the list of unchecked hashes
+to be checked in the next cycle. Once all hashes in the current cycle have been iterated through,
+the previous end block + 1 becomes the start block for the next cycle, and the latest block is
+the new end block. Daemon sleeps for 30 mins and continue checking.
+"""
+import time
+from typing import List
+from web3 import Web3
+from onchain import OnChainEBBO
+
+INFURA_KEY = '625ae5b0766246c5bed343445131cd2c'
+class DaemonEBBO:
+    """
+    Initialization of EBBOAnalysis class object and logger object.
+    """
+
+    def __init__(self):
+        self.instance = OnChainEBBO()
+        # self.logger = get_logger()
+
+    def main(self, sleep_time: int) -> None:
+        """
+        daemon function that runs as highlighted in docstring.
+        """
+        infura_connection = f"https://mainnet.infura.io/v3/{INFURA_KEY}"
+        web_3 = Web3(Web3.HTTPProvider(infura_connection))
+        start_block = web_3.eth.block_number
+        print("starting...")
+        unchecked_hashes: List[str] = []
+        while True:
+            time.sleep(sleep_time)
+            end_block = web_3.eth.block_number
+            fetched_hashes = self.instance.get_hashes_by_block(start_block, end_block)
+            all_hashes = fetched_hashes + unchecked_hashes
+            unchecked_hashes = []
+            while len(all_hashes) > 0:
+                single_hash = all_hashes.pop(0)
+                if self.onchain_quasimodo_test(single_hash) == None:
```

Comment:
It looks like `onchain_quasimodo_test` returns a boolian and that condition would be `false` always. In any case, this logic could be moved into the function itself.

Reply:
It won't always be false. 
If order fails to be parsed, None is returned, which evaluates to True in this condition.
Changed the main loop line                
` if self.onchain_quasimodo_test(single_hash) == None:`
to 
`if self.onchain_quasimodo_test(single_hash) == True:`

So, if None=None, returns true and adds to list of unchecked hashes.
In all other cases, it's false and hash is removed from list.

Reply:
The expression `self.instance.decode_single_hash(single_hash) is None` below in the definition of `onchain_quasimodo_test` looks as if it compares to `None` there. Then this comparison would give `True` or `False`. And later that result would be compared to `None` again. (But my python is not good. Might just be syntax not intuitive to me.)

Reply:
If the value is always `True` or `False` you could just leave out the comparison instead of comparing to `True`.